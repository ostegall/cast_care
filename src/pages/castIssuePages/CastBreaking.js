import { Text, View, Pressable } from 'react-native';
import styles from '../../styles';

export default function CastBreaking({ navigation }) { //Solutions for user that has a breaking/broken cast
  return (
    <View style={{ flex: 1, paddingTop: 20 }}>
      <Text style={styles.header_text}>
        Problem Description:
      </Text>
      <Text style={styles.page_text}>
        Is patient's cast breaking or soft in certain places?
      </Text>
      <Text style={styles.header_text}>
        Why this is a Problem:
      </Text>
      <Text style={styles.page_text}>
        This could indicate the cast is not thick enough to support patient's fracture.
      </Text>
      <Text style={styles.header_text}>
        Correcting the Problem:
      </Text>
      <Text style={styles.page_text}>
        It is important to be assessed to get another cast put on to properly support the fracture.
      </Text>
      <Text style={styles.header_text}>
        Who to Contact:
      </Text>
      <Text style={styles.page_text}>
        Patient needs to come into clinic the same day or next day for evaluation. To do so, please call 984-974-5700 to schedule an appointment.
      </Text>
      <View style={{ alignItems: 'center', justifyContent: 'center' }}>
        <View style={styles.buttonRow}>
          <Pressable style={styles.button}
            onPress={() => {
              navigation.navigate("Info") //Navigates to page for finding nearest hospital
            }}
          >
            <Text style={styles.text}>
              Find your nearest location
            </Text>
          </Pressable>
          <Pressable style={styles.button}
            onPress={() => {
              navigation.navigate("Cast Breaking Blur") //Navigates to blurred image of problem
            }}
          >
            <Text style={styles.text}>
              Show image of breaking cast
            </Text>
          </Pressable>
        </View>
      </View>
    </View>
  )
}