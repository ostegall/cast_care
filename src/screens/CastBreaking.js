import React from 'react';
import { Text, View } from 'react-native';

class CastBreaking extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <View>
        <Text>Cast Breaking? Selection
        </Text>
      </View>
    )
  }
}

export default CastBreaking