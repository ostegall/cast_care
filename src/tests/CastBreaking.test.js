import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'

import CastBreaking from '../pages/castIssuePages/CastBreaking';

//Makes sure app component renders correctly without crashing
it("renders without crashing", () => {
  shallow(<CastBreaking />);
});

//Snapshot test to ensure UI stays consistent
it('renders correctly', () => {
  const tree = renderer.create(<CastBreaking />).toJSON();
  expect(tree).toMatchSnapshot();
});