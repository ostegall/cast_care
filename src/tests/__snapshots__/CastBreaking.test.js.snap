// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
        "letterSpacing": 0.5,
        "textAlign": "center",
      }
    }
  >
    Problem Description:
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    Is patient's cast breaking or soft in certain places?
  </Text>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
        "letterSpacing": 0.5,
        "textAlign": "center",
      }
    }
  >
    Why this is a Problem:
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    This could indicate the cast is not thick enough to support patient's fracture.
  </Text>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
        "letterSpacing": 0.5,
        "textAlign": "center",
      }
    }
  >
    Correcting the Problem:
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    It is important to be assessed to get another cast put on to properly support the fracture.
  </Text>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
        "letterSpacing": 0.5,
        "textAlign": "center",
      }
    }
  >
    Who to Contact:
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    Patient needs to come into clinic the same day or next day for evaluation. To do so, please call 984-974-5700 to schedule an appointment.
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "height": 83,
          "marginLeft": 24,
          "marginRight": 23,
          "marginTop": 20,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4B9CD3",
            "borderRadius": 10,
            "elevation": 2,
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "center",
            "minWidth": 88,
            "paddingLeft": 16,
            "paddingRight": 16,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.35,
            "shadowRadius": 5,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "letterSpacing": 0.25,
              "textAlign": "center",
            }
          }
        >
          Find your nearest location
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4B9CD3",
            "borderRadius": 10,
            "elevation": 2,
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "center",
            "minWidth": 88,
            "paddingLeft": 16,
            "paddingRight": 16,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.35,
            "shadowRadius": 5,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "letterSpacing": 0.25,
              "textAlign": "center",
            }
          }
        >
          Show image of breaking cast
        </Text>
      </View>
    </View>
  </View>
</View>
`;
